packages/react-components/react-carousel/library/src/components/useEmblaCarousel.ts

```diff
@@ -52,6 +52,9 @@ export function useEmblaCarousel(
     initialState: 0,
   });
 
+  const indexRef = React.useRef<number>(activeIndex);
+  indexRef.current = activeIndex;
```

Comment:
> Don’t read or write ref.current during rendering. If some information is needed during rendering, use [state](https://react.dev/learn/state-a-components-memory) instead. Since React doesn’t know when ref.current changes, even reading it while rendering makes your component’s behavior difficult to predict.

https://react.dev/learn/referencing-values-with-refs#best-practices-for-refs

Reply:
We don't want to use state here as we shouldn't be driving index changes on update, this is purely to detect when the controlled index and rendered index becomes out of sync due to setting to a newly added card once the slidesChanged callback occurs (engine cannot set to out of bounds index before new card is rendered and registered).

This should be a rare occurrence but will fix the issue of trying to set index to a newly rendered card.

Alternatively we can expose the slides changed callback and let users handle this, but it would be nice to enable this internally as we are receiving feedback this may be a common usecase.

Reply:
I'll check if we can place this in a useEffect dependent on activeIndex instead to remove from our render cycle.